### PR TITLE
Week 2 Assessment completed and ready for review

### PR DESCRIPTION
**Pull request by aprilrieger**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -1,63 +1,106 @@
...
 
 var num1 = 15
+
+const myFunction = () => {
+    if(num1 % 3 === 0){
+        return `${num1} is divisible by three`
+    } else {
+        return `${num1} is not divisible by three`
+    }
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:14]: how can we do this with just a single function? hint: passing the numbers in as an argument

File: code-challenges.js [hunk 2]
```diff
@@ -1,63 +1,106 @@
...
-
-
+let newArr = []
 
+const myFunction = (array) => {
+    let newString = array.map(value => {
+      return value[0].toUpperCase() + value.substring(1)
+    })
+     return newString
+  }
```
Comment (rachaelstanislaw) [inline code-challenges.js:54]: great job!

File: code-challenges.js [hunk 3]
```diff
@@ -1,63 +1,106 @@
...
 
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
-// Expected output: [-8, 0, 8, 46, 59, 90, 107]
-
-
 
+const numsOnly = (arr) => {
+	let nums = arr.filter(element => typeof element === "number").sort((a,b) => a - b);
+	return nums;
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:66]: woohoo!

File: code-challenges.js [hunk 4]
```diff
@@ -1,63 +1,106 @@
...
-
-
-
+const firstVowelIndex = (string) => {
+	let vowels = "aeiou";
+	for(let i = 0; i < string.length; i++){
+		if(vowels.includes(string[i]))
+			return i;
+	}
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:83]: wonderful!

File: code-challenges.js [hunk 5]
```diff
@@ -1,63 +1,106 @@
...
-
+const calculator = (num1, exp, num2) => {
+	if(exp === "+") return num1+num2;
+	else if(exp === "-") return num1 - num2;
+	else if(exp === "*") return num1 * num2;
+	else {
+		if(num2 === 0) return "Can't divide by 0!"
+		else return num1 / num2;
+	}
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:97]: yay!!

File: interview-questions.md [hunk 6]
```diff
@@ -1,62 +1,76 @@
...
 
 Answer the following questions. First, without external resources. Challenge yourself to answer from memory. Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn.
 
 1. Why would you use the method super()?
 
-  Your answer:
+  Your answer: We are able to inherit from the parent
 
-  Researched answer:
+  Researched answer: The super() method refers to the parent class. By calling the super() method in the constructor method, we call the parent's constructor method and gets access to the parent's properties and methods. https://www.w3schools.com/js/js_classes.asp
```
Comment (rachaelstanislaw) [inline interview-questions.md:9]: good research!

File: interview-questions.md [hunk 7]
```diff
@@ -1,62 +1,76 @@
...
 
 
 
 2. What is a virtual DOM?
 
-  Your answer:
+  Your answer: I don't know the definition of virtual DOM but DOM stands for the Document Object Model which n interface, so taking a stab that this is a virtual interface.
```
Comment (rachaelstanislaw) [inline interview-questions.md:15]: good guess!

File: interview-questions.md [hunk 8]
```diff
@@ -1,62 +1,76 @@
...
 
-  Researched answer:
+  Researched answer: In React, for every DOM object, there is a corresponding “virtual DOM object.” A virtual DOM object is a representation of a DOM object, like a lightweight copy.
+
+  A virtual DOM object has the same properties as a real DOM object, but it lacks the real thing’s power to directly change what’s on the screen.
+
+  Manipulating the DOM is slow. Manipulating the virtual DOM is much faster, because nothing gets drawn onscreen. Think of manipulating the virtual DOM as editing a blueprint, as opposed to moving rooms in an actual house.
+
+  https://www.codecademy.com/articles/react-virtual-dom
```
Comment (rachaelstanislaw) [inline interview-questions.md:23]: good research again!!

File: interview-questions.md [hunk 9]
```diff
@@ -1,62 +1,76 @@
...
-  Your answer:
+  Your answer: You will need a function that returns the JSX value in the H1 tags (see example below).
+  class Function = props => {
+      <div><h1>Hello World</h1></div>
+    }
+  }
+  export default App
 
-  Researched answer:
+  Researched answer: A basic stateless component is just a plain javascript function which takes props as an argument and returns a react element.
```
Comment (rachaelstanislaw) [inline interview-questions.md:36]: for a functional component, yes!

File: interview-questions.md [hunk 10]
```diff
@@ -1,62 +1,76 @@
...
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to display "Hello World" in the browser?
 
-  Your answer:
+  Your answer: You will need a function that returns the JSX value in the H1 tags (see example below).
+  class Function = props => {
+      <div><h1>Hello World</h1></div>
+    }
+  }
+  export default App
```
Comment (rachaelstanislaw) [inline interview-questions.md:34]: syntax is a little off here but i see what you're going for

File: interview-questions.md [hunk 11]
```diff
@@ -1,62 +1,76 @@
...
+    Updating - When something in the component is updated (ie state)
+    Unmounting - When a component is removed from the DOM
+- API - a set of functions and procedures allowing the creation of applications that access the features or data of an operating system, application, or other service.
+- event.preventDefault() - A method that tells the user agent that if the event does not get explicitly handled, its default action should not be taken as it normally would be.
+- DOM events - are actions that occurs as a result of the user action or as result of state change of the elements of a DOM tree.
+- http verbs - The primary or most-commonly-used HTTP verbs (or methods, as they are properly called) are POST, GET, PUT, PATCH, and DELETE. These correspond to create, read, update, and delete (or CRUD) operations, respectively. There are a number of other verbs, too, but are utilized less frequently.
+- MVC - MVC stands for Model-View-Controller. It's a design pattern that breaks an application into three parts: the data (Model), the presentation of that data to the user (View), and the actions taken on any user interaction (Controller).
```
Comment (rachaelstanislaw) [inline interview-questions.md:76]: good job on these short answers!

File: react-challenge.html [hunk 12]
```diff
@@ -66,25 +64,23 @@ <h1>Welcome to the Mood Meter App</h1>
...
               <p style={{fontSize: "40px"}}>{ this.props.moodOptions }</p>
-              <button onClick={ }>Click Here For Next Mood Option</button>
+              <button onClick={ this.props.next }>Click Here For Next Mood Option</button>
               <br />
               <br />
               <h3>Save your mood here!</h3>
               <button onClick={ this.props.setThisMood }>Yup, this is me right now.</button>
               <h3>My mood:</h3>
               <p style={{fontSize: "40px"}}>{ this.props.set }</p>
+            </div>
```
Comment (rachaelstanislaw) [inline react-challenge.html:77]: yay!! great job on this!

**Review by rachaelstanislaw — APPROVED** [review]
yay! week 2 in the bag! give me a heads up after you read feedback and let me know if you have any questions before you merge :) good work!

File: code-challenges.js (hunk 1 above)
Comment (aprilrieger) [reply code-challenges.js:14]: Hi Rachael,
On review of our assessment last Monday morning I realized that the prompt was different than I had thought. So if I were to pass all three numbers into a single function I would do it like this: 

const myFunction = (number) => {
    if(number % 3 ===0){
        return `${number} is divisible by 3.`
    } else if(number% 3 !== 0){
        return `${number} is not divisible by 3.`
    } else {
        return "error"
    }
}
console.log(myFunction(num1))
console.log(myFunction(num2))
console.log(myFunction(num3))

File: interview-questions.md (hunk 9 above)
Comment (aprilrieger) [reply interview-questions.md:36]: After today's lecture I understand the difference between a dumb and smart component a bit better :)

File: interview-questions.md (hunk 10 above)
Comment (aprilrieger) [reply interview-questions.md:34]: We would need a render, return and a React.Fragment or <></> its all starting to make more sense each day :) YAY!

**Comment by rachaelstanislaw** [comment]
Fantastic! You’re good to merge :)

**Pull request merged**